DEV-686 Table component redesign with light/dark mode

willmccormick24 opened the pull request:
I added the ability to sort the data by headers alphanumerically. Provided simple mock data to visualize table.

Light Mode:
![Screenshot 2023-09-19 at 12 13 00 PM](https://github.com/rebuschain/rebus.app.web/assets/143646590/b5a71ec0-14de-4ba3-8623-73579694b63a)

Dark Mode:
![Screenshot 2023-09-19 at 12 12 09 PM](https://github.com/rebuschain/rebus.app.web/assets/143646590/aa1e6391-3eb2-43e6-adad-8be30d06b04b)

Ericky14 commented on src/components/common/table.tsx:
```diff
@@ -0,0 +1,96 @@
+	data: any[];
+}
+
+const Table: React.FC<TableProps> = ({ headers, data }) => {
```
Please don't make your own table, table components are complicated and we already have a table component

willmccormick24 commented:
The new table also reskins the Assets Page, which is the only implementation of a table after the reskin.

Light Mode:
![Screenshot 2023-09-28 at 5 47 59 PM](https://github.com/rebuschain/rebus.app.web/assets/143646590/0c8fde5b-00d3-4306-86ae-a9b38f12602c)

Dark Mode:
![Screenshot 2023-09-28 at 5 47 33 PM](https://github.com/rebuschain/rebus.app.web/assets/143646590/9747536c-5ea7-45af-ac37-4274e54cf7d8)

nicolam1 commented on src/colors.tsx:
```diff
@@ -0,0 +1,21 @@
+//Colors for Redesign
+
+const colors = {
```
I keep seeing this file in the PRs. I think you are pointing to the wrong branch maybe

willmccormick24 replied:
@nicolam1 that's the new color scheme for the redesign I made initially with the new colors from FIgma. It's referenced in the `src/theme.tsx` file just to make the code a little more readable. I can get rid of it and just put the colors directly in the theme file if you want? The `colors.tsx` file is already in master though, this branch is able to merge with master.

Ericky14 replied:
@willmccormick24 If this was in master then we wouldn't see this change in the DIFF, maybe rebase the branch on master and push again

nicolam1 reviewed (changes requested):
Fix commit or PR

Ericky14 commented on src/pages/stake/table.tsx:
```diff
@@ -297,11 +297,13 @@ const Table = observer<TableProps>(({ active, className, tableRowClassName }) =>
 	const noDataFound = <div className="no_data_table"> No data found </div>;
 	const loader = <CircularProgress />;
 
+	const data = [['Test'], ['Test'], ['Test']];
```
Remove?

Ericky14 reviewed (changes requested):
You messed up styles in other places in the app

<img width="1114" alt="Screenshot 2023-10-02 at 11 22 48 AM" src="https://github.com/rebuschain/rebus.app.web/assets/21248862/2e3fa943-07a6-467a-bacf-ebd9c8794a03">
<img width="1386" alt="Screenshot 2023-10-02 at 11 23 56 AM" src="https://github.com/rebuschain/rebus.app.web/assets/21248862/f232f750-9e06-44e1-b710-09fe87fd0f50">

willmccormick24 commented:
In the most recent commit, I addressed the comments and I fixed the button issues in DEV-672 pages redesign then made this branch point to that, so should be good to go 👍
